opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -803,4 +803,134 @@ mod tests {
             "Metrics should be available in exporter."
         );
     }
+
+    async fn some_async_function() -> u64 {
```

Comment:
@howardjohn If you have bandwidth, could you review these? Asking since you helped fix some bugs in the area previously!